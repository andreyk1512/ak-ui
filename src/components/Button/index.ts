export * from './Button';
export * from './Button.types';

import { Link } from './components';
import { Button as OriginButton } from './Button';

export type ButtonProps = typeof OriginButton & {
  Link: typeof Link;
};

export const Button = OriginButton as ButtonProps;
Button.Link = Link;
