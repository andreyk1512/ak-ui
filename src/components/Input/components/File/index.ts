export * from './File';
export * from './File.types';
