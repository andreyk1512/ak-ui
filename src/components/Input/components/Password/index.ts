export * from './Password';
export * from './Password.types';
