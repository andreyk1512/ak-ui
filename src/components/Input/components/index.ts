export * from './Password';
export * from './TextArea';
export * from './File';
