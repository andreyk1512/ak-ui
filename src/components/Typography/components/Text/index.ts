export * from './Text.tsx';
export * from './Text.types';
