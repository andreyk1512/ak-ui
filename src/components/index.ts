export * from './Tag';
export * from './Button';
export * from './Checkbox';
