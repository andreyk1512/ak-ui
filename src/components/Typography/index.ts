export * from './Typography';
export * from './Typography.types';

import { Text, Title, Paragraph } from './components';
import { Typography as OriginTypography } from './Typography';

export type TypographyProps = typeof OriginTypography & {
  Text: typeof Text;
  Title: typeof Title;
  Paragraph: typeof Paragraph;
};

export const Typography = OriginTypography as TypographyProps;
Typography.Text = Text;
Typography.Title = Title;
Typography.Paragraph = Paragraph;
