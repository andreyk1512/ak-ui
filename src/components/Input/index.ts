export * from './Input';
export * from './Input.types';

import { Password, TextArea, File } from './components';
import { Input as OriginInput } from './Input';

export type InputProps = typeof OriginInput & {
  Password: typeof Password;
  TextArea: typeof TextArea;
  File: typeof File;
};

export const Input = OriginInput as InputProps;
Input.Password = Password;
Input.TextArea = TextArea;
Input.File = File;
