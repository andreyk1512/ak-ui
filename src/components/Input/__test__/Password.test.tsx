import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Password } from '../components';

describe('Input.Password', () => {
  it('should render the input element with type password', () => {
    const { container } = render(<Password aria-label="password" />);

    const input = screen.getByLabelText('password');
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'password');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should change type to text when click on icon', () => {
    const { container } = render(<Password aria-label="password" />);

    const visibilityIcon = screen.getByLabelText('visibility-icon');
    fireEvent.click(visibilityIcon);
    const input = screen.getByLabelText('password');
    expect(input).toHaveAttribute('type', 'text');
    expect(container.firstChild).toMatchSnapshot();
  });
});
