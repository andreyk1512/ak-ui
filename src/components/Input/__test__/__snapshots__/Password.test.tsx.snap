// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.Password should change type to text when click on icon 1`] = `
<div
  class="css-1i5hjp5"
>
  <input
    aria-label="password"
    class="css-tws2u3"
    type="text"
  />
  <span
    class="css-1hbtery"
  >
    <span
      aria-label="visibility-icon"
      class="css-xbmx3s"
    >
      <span />
    </span>
  </span>
</div>
`;

exports[`Input.Password should render the input element with type password 1`] = `
<div
  class="css-1i5hjp5"
>
  <input
    aria-label="password"
    class="css-tws2u3"
    type="password"
  />
  <span
    class="css-1hbtery"
  >
    <span
      aria-label="visibility-icon"
      class="css-xbmx3s"
    >
      <span />
    </span>
  </span>
</div>
`;
